### PR TITLE
feat(k8s): add namespace for api jobs and add secrets to both namespaces

### PR DESCRIPTION
**Pull request by m90**
Requires: https://github.com/wmde/wbaas-deploy/pull/892

Ticket: https://phabricator.wikimedia.org/T330389

As discussed, we want to create a dedicated namespace that contains the rather noisy job objects. As there seems to be no official way of sharing secrets across namespaces, I went for defining the namespace in terraform and also duplicating the secrets across namespaces.

~~Leaving this a draft as it contains dev junk still.~~

### CODE REVIEW

File: tf/env/local/namespaces.tf [hunk 1]
```diff
@@ -0,0 +1,5 @@
+resource "kubernetes_namespace" "api-job-namespace" {
+  metadata {
+    name = "api-jobs"
+  }
+}
```
Comment (deer-wmde) [inline tf/env/local/namespaces.tf:5]: why is this needed for local / not needed for staging & prod?

Comment (m90) [reply tf/env/local/namespaces.tf:5]: It should also go into staging and prod, I just did not finish this PR yet :)

Comment (deer-wmde) [reply tf/env/local/namespaces.tf:5]: ahh whoops sorry :sweat_smile:

File: tf/env/local/secrets-api.tf [hunk 2]
```diff
@@ -29,13 +29,14 @@ resource "random_password" "api-app-jwt-secret" {
 }
 
 resource "kubernetes_secret" "api-app-secrets" {
+  for_each = toset(["default", "api-jobs"])
   metadata {
-    name = "api-app-secrets"
-    namespace = "default"
+    name      = "api-app-secrets"
+    namespace = each.value
```
Comment (m90) [inline tf/env/local/secrets-api.tf:35]: While `terraform` can `plan` this change properly, it seems dropping and recreating a secret like this _sometimes_ creates race conditions in Kubernetes, making tf fail at apply-time. Not sure what to do about it right now to be honest.

Comment (m90) [reply tf/env/local/secrets-api.tf:35]: This can be fixed using `moved` blocks.

**Review by deer-wmde — APPROVED** [review]
Can confirm this worked locally for me! details: https://phabricator.wikimedia.org/T330389#8895083

**Review by deer-wmde — APPROVED** [review]
Can confirm this worked locally for me! details: https://phabricator.wikimedia.org/T330389#8895083
